machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,117 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfig is the Schema for the nutanixmachineproviderconfigs API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// clusterIdentifier is to identify the cluster (the Prism Element under management
+	// of the Prism Central), in which the Machine's VM will be created.
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	ClusterIdentifier NutanixResourceIdentifier `json:"clusterIdentifier"`
+
+	// imageIdentifier is to identify the rhcos image uploaded to the Prism Central (PC)
+	// The image identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	ImageIdentifier NutanixResourceIdentifier `json:"imageIdentifier"`
+
+	// subnetIdentifier is to identify the cluster's network subnet to use for the Machine's VM
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	SubnetIdentifier NutanixResourceIdentifier `json:"subnetIdentifier"`
```

Comment:
IMO we don't need any suffix on these fields, I would just make them `Cluster`, `Image` and `Subnet`, are you keen to have some sort of suffix on these? Typically we wouldn't have any suffix if you look at other provider APIs